src/vs/vscode.d.ts

```diff
@@ -4325,6 +4330,29 @@ declare module 'vscode' {
 		resolveTask(task: Task, token?: CancellationToken): ProviderResult<Task>;
 	}
 
+	export enum LogLevel {
+		Trace = 1,
+		Debug = 2,
+		Info = 3,
+		Warning = 4,
+		Error = 5,
+		Critical = 6,
+		Off = 7
+	}
+
+	export interface ILogger {
+		onDidChangeLogLevel: Event<LogLevel>;
+		getLevel(): LogLevel;
+		getLogDirectory(): Thenable<string>;
```

Comment:
https://github.com/Microsoft/vscode/wiki/Extension-API-guidelines#sync-vs-async

Reply:
also, no getter, just a property

Reply:
It's async because it only creates the directory when requested. It could be a prop of a promise though.